.github/workflows/ci.yaml

```diff
@@ -0,0 +1,92 @@
+name: Run-Tests
+
+on:
+  pull_request: {}
+  push:
+    branches:
+      - main
+
+jobs:
+  fetch-lib-check:
+    name: Check the charm uses up-to-date charm libraries
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check libs
+      uses: canonical/charm-lib-checker-action@main
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+  lib-check:
+    name: Static analysis of /lib for Python 3.5
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Set up Python 3.5
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.5
+    - name: Install dependencies
+      run: python3 -m pip install tox
+    - name: Run static analysis for /lib for 3.5
+      run: tox -vve static-lib
+  static-analysis:
+    name: Static analysis
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Install dependencies
+      run: python3 -m pip install tox
+    - name: Run static analysis (charm)
+      run: tox -vve static-charm
+    - name: Run static analysis (unit tests)
+      run: tox -vve static-unit
+    - name: Run static analysis (integration tests)
+      run: tox -vve static-integration
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: [3.8]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Setup Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python }}
+    - name: Install dependencies
+      run: python3 -m pip install tox
+    - name: Run linters
+      run: tox -vve lint
+  unit-test:
+    name: Unit tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.8]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Set up python ${{ matrix.python }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python }}
+    - name: Install dependencies
+      run: python -m pip install tox
+    - name: Run tests
+      run: tox -e unit
+  integration-test-microk8s:
+    name: Integration tests (microk8s)
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Setup operator environment
+      uses: charmed-kubernetes/actions-operator@main
+      with:
+        juju-channel: latest/candidate # FIXME: Once 2.9.23 hits stable
+        provider: microk8s
+    - name: Run integration tests
+      run: tox -vve integration
```

Comment:
```suggestion
#  integration-test-microk8s:
#    name: Integration tests (microk8s)
#    runs-on: ubuntu-latest
#    steps:
#    - name: Checkout
#      uses: actions/checkout@v2
#    - name: Setup operator environment
#      uses: charmed-kubernetes/actions-operator@main
#      with:
#        juju-channel: latest/candidate # FIXME: Once 2.9.23 hits stable
#        provider: microk8s
#    - name: Run integration tests
#      run: tox -vve integration
```

Disable until we have any integration tests or the CI will continue to fail. 😅

Reply:
Actually will need to wait until they _also_ support machine charms whenever we have time for the "big" bundle framework